Makefile

```diff
@@ -0,0 +1,16 @@
+all: doc example demo
+
+paq:
+	cd src && ./paq.sh > ../nuklear.h
+
+doc: paq
+	$(MAKE) -C doc
+	cd doc && ./build.sh
+
+example: paq
```

Comment:
Building example(s) shouldn't have `paq` as dependency as the current `nuklear.h` is always in this upstream repo. I understand that when testing, one needs to rebuild e.g. some demos, but that's not the point of this general Makefile and definitely not the standard use case of `examples/`.